src/hooks/use-settings.js

```diff
@@ -79,7 +76,32 @@ const useSettings = ({ core, models, flags }) => {
     chartModel.command.layoutComponents({ settings: newSettings });
     updateViewState({ viewState, viewStateOptions: options.viewState, models });
     setSettings(newSettings);
-  }, [rect.width, rect.height, constraints]);
+  };
+
+  useEffect(() => {
+    if (!models || !models?.colorService.isInitialized()) {
+      return;
+    }
+
+    const { chartModel } = models;
+    if (objectHelper.isEqual(constraints, chartModel.query.getMeta().previousConstraints)) {
```

Comment:
you could probably do the following instead of adding this new isEqual utility func:
```
JSON.stringify(constraints) === JSON.stringify(chartModel.query.getMeta().previousConstraints)
```

Reply:
interesting! I'll try.